withastro/astro
# Actions: Add `devalue` for serializing complex values

## Changes

Standardize our Actions serialization strategy. This introduces `devalue` to allow complex values like Dates and Sets, and adds serialization to middleware results for edge middleware support.

- Abstract serialization and deserialization (with REST response metadata) to `serializeActionResult` and `deserializeActionResult` utils
- Pass serializable values from middleware locals to ensure edge middleware works as expected
- Update frontend fetch to user deserializer

## Testing

- Add integration test for Date and Set
- Update expected headers and value checks to use devalue

## Docs

N/A - should continue to work as expected. The RFC did not specify supported values
